src/rust/storage/seg/src/hashtable/mod.rs

```diff
@@ -541,55 +520,35 @@ impl HashTable {
         let tag = tag_from_hash(hash);
         let bucket_id = hash & self.mask;
 
-        let mut bucket = &mut self.data[bucket_id as usize];
-        let chain_len = chain_len(bucket.data[0]);
-
-        for chain_idx in 0..=chain_len {
-            let n_item_slot = if chain_idx == chain_len {
-                N_BUCKET_SLOT
-            } else {
-                N_BUCKET_SLOT - 1
-            };
-            for i in 0..n_item_slot {
-                if chain_idx == 0 && i == 0 {
-                    continue;
-                }
-                let current_info = bucket.data[i];
+        let iter = IterMut::new(self, hash);
 
-                if get_tag(current_info) == tag {
-                    let current_item = segments.get_item(current_info).unwrap();
-                    if current_item.key() != key {
-                        HASH_TAG_COLLISION.increment();
-                    } else {
-                        // update item frequency
-                        let mut freq = get_freq(current_info);
-                        if freq < 127 {
-                            let rand: u64 = self.rng.gen();
-                            if freq <= 16 || rand % freq == 0 {
-                                freq = ((freq + 1) | 0x80) << FREQ_BIT_SHIFT;
-                            } else {
-                                freq = (freq | 0x80) << FREQ_BIT_SHIFT;
-                            }
-                            if bucket.data[i] == current_info {
-                                bucket.data[i] = (current_info & !FREQ_MASK) | freq;
-                            }
-                        }
-
-                        if cas == get_cas(bucket.data[0]) {
-                            // TODO(bmartin): what is expected on overflow of the cas bits?
```

Comment:
Good catch @thinkingfish - Per discussion this is fixed as part of #413, comment should be removed